File: tests/ctest/test_format_mismatch.cpp
#include <vmdio/model_edit.h>
#include <vmdio/camera_edit.h>
#include <vmdio/vmd_exceptions.h>

#include "test_base.h"

namespace vmd_model = vmdio::model_edit;
namespace vmd_camera = vmdio::camera_edit;
namespace vmd_except = vmdio::exceptions;

class FormatMismatchTest : public test_base::TempVmdFileTest
{
};

TEST_F(FormatMismatchTest, ReadCameraEditVmdWithModelEditReader)
{
    std::filesystem::path lCameraVmdPath = mTestMotionDataDir / "camera.vmd";

    EXPECT_THROW(vmd_model::readVMD(lCameraVmdPath), vmd_except::IncompatibleFormatError);
}

TEST_F(FormatMismatchTest, WriteCameraEditModelNameWithModelEditWriter)
{
    vmd_model::VMDData lBadData;
    lBadData.modelName = test_base::makeVMDString(u8"カメラ・照明");

    EXPECT_THROW(
        vmd_model::writeVMD(lBadData, mTempFilePath), vmd_except::IncompatibleFormatError);
}

TEST_F(FormatMismatchTest, ReadModelEditVmdWithCameraEditReader)
{
    std::filesystem::path lModelEditVmdPath = mTestMotionDataDir / "motion_morph_visibleik.vmd";

    EXPECT_THROW(vmd_camera::readVMD(lModelEditVmdPath), vmd_except::IncompatibleFormatError);
}

File: tests/pytest/test_format_mismatch.py
import pytest

import pyvmdio.model_edit as vmdio_model
import pyvmdio.camera_edit as vmdio_camera
import pyvmdio.exceptions as vmdio_exceptions
from pyvmdio.vmd_string import VMDString


def test_read_camera_edit_vmd_as_model_edit(test_motion_data_dir):
    src = test_motion_data_dir / "camera.vmd"
    with pytest.raises(vmdio_exceptions.IncompatibleFormatError):
        vmdio_model.readVMD(src)


def test_write_camera_edit_vmd_from_model_edit(temp_vmd_path):
    dummy = vmdio_model.VMDData()
    dummy.modelName = VMDString.fromUTF8("カメラ・照明")

    with pytest.raises(vmdio_exceptions.IncompatibleFormatError):
        vmdio_model.writeVMD(dummy, temp_vmd_path)


def test_read_model_edit_vmd_as_camera_edit(test_motion_data_dir):
    src = test_motion_data_dir / "motion_morph_visibleik.vmd"
    with pytest.raises(vmdio_exceptions.IncompatibleFormatError):
        vmdio_camera.readVMD(src)